Sources/PackageModel/ToolsVersion.swift

```diff
@@ -36,12 +36,12 @@ public struct ToolsVersion: Equatable, Hashable, Codable {
     /// Regex pattern to parse tools version. The format is SemVer 2.0 with an
     /// addition that specifying the patch version is optional.
     static let toolsVersionRegex = try! NSRegularExpression(pattern: "^" +
-        "(\\d+)\\.(\\d+)(?:\\.(\\d+))?" +
+        #"(\d+)\.(\d+)(?:\.(\d+))?"# +
         "(" +
-            "\\-[A-Za-z\\d]+(?:\\.[A-Za-z\\d]+)*" +
+            #"\-[A-Za-z\d]+(?:\.[A-Za-z\d]+)*"# +
         ")?" +
         "(" +
-            "\\+[A-Za-z\\d]+(?:\\.[A-Za-z\\d]+)*" +
+            #"\+[A-Za-z\d]+(?:\.[A-Za-z\d]+)*"# +
         ")?$", options: [])
```

Comment:
If we're going through the exercise of rewriting this, any reason not use a multi-line, raw string literal with the [`. allowCommentsAndWhitespace`](https://developer.apple.com/documentation/foundation/nsregularexpression/options/1413559-allowcommentsandwhitespace) option?

```swift
static let toolsVersionRegex = try! NSRegularExpression(pattern: #
^(\d+)\.(\d+)(?:\.(\d+))?
(
  \-[A-Za-z\d]+(?:\.[A-Za-z\d]+)*
)?
(
  \+[A-Za-z\d]+(?:\.[A-Za-z\d]+)*
)?$
"""#, options: [.allowCommentsAndWhitespace])

Reply:
This is a good point. I originally thought about using a multi-line literal, but didn't do it because I forgot about using `.allowCommentsAndWhitespace` to retain the "scoping". Thanks for reminding me about it!

I changed the pattern to using a multi-line literal in commit 1e24af8a3d30409afcc5a1aac6050ed7b955f1da, with a bit of stylistic differences.